fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -188,10 +187,55 @@ public HiveTableStats getTableStats() throws DdlException {
         return result;
     }
 
-    public void refreshTableCache() throws DdlException {
-        Catalog.getCurrentCatalog().getHiveRepository()
-                .refreshTableCache(resourceName, hiveDb, hiveTable, getPartitionColumns(),
-                        new ArrayList<>(nameToColumn.keySet()));
+    public void refreshTableCache(String dbName, String tableName) throws DdlException {
+        org.apache.hadoop.hive.metastore.api.Table table = Catalog.getCurrentCatalog().getHiveRepository()
+                .getTable(resourceName, this.hiveDb, this.hiveTable);
+        List<FieldSchema> unPartHiveColumns = table.getSd().getCols();
+        List<FieldSchema> partHiveColumns = table.getPartitionKeys();
+        Map<String, FieldSchema> allHiveColumns = unPartHiveColumns.stream()
+                .collect(Collectors.toMap(FieldSchema::getName, fieldSchema -> fieldSchema));
+        for (FieldSchema hiveColumn : partHiveColumns) {
+            allHiveColumns.put(hiveColumn.getName(), hiveColumn);
+        }
+        boolean needRefreshColumn = allHiveColumns.size() != this.fullSchema.size();
+        if (!needRefreshColumn) {
+            for (Column column : fullSchema) {
+                FieldSchema fieldSchema = allHiveColumns.get(column.getName());
```

Comment:
I think it can't guarantee that the two columns are exactly the same through judging only by the column name